test/service/test_response.py

```diff
@@ -2190,6 +2181,41 @@ def _test(entity_type: str, expect_empty: bool, expect_accessible: bool):
             for entity_type, is_filtered in filtered_entity_types.items():
                 _test(entity_type, expect_empty=is_filtered, expect_accessible=False)
 
+    def test_filter_by_accession(self):
+        def request_accessions(nested_properties):
+            params = self._params(filters={
+                'accessions': {
+                    'is': [nested_properties]
+                }
+            })
+            url = self.base_url.set(path='/index/projects')
+            response = requests.get(str(url), params=params)
+            self.assertEqual(200, response.status_code)
+            return response.json()
+
+        for nested_properties in [
+            dict(namespace='array_express', accession='E-AAAA-00'),
+            dict(namespace='geo_series', accession='GSE132044'),
+        ]:
+            with self.subTest(nested_properties=nested_properties):
+                hit = one(request_accessions(nested_properties)['hits'])
+                self.assertIn(nested_properties, one(hit['projects'])['accessions'])
+                del nested_properties['namespace']
+                self.assertEqual(hit, one(request_accessions(nested_properties)['hits']))
+
+        for case in [
+            dict(nested_properties=dict(namespace='geo_series'),
+                 project_and_bundle=[('627cb0ba', 'fa5be5eb'), ('88ec040b', 'ffac201f')])
+        ]:
+            with self.subTest(nested_properties=nested_properties):
+                response = request_accessions(case['nested_properties'])
+                expected_projects = {'627cb0ba-b8a1-405a-b58f-0add82c3d635', '88ec040b-8705-4f77-8f41-f81e57632f7d'}
+                projects = {
+                    one(project['projectId'])
+                    for project in response['termFacets']['project']['terms']
```

Comment:
```suggestion
                    one(hit['projects'])['projectId']
                    for hit in response['hits']
```
I feel it's a stronger test if the values are pulled from the hits instead of the termFacets.